test/subscription_map.hpp

```diff
@@ -487,37 +549,36 @@ class multiple_subscription_map
         auto insert_result = subscription_set.insert_or_assign(std::forward<K>(key), std::forward<V>(value));
         if(insert_result.second) {
             this->increase_subscriptions(h);
-            ++this->map_size;
+            this->increase_map_size();
         }
-        return std::make_pair(std::move(h), insert_result.second);
+        return std::make_pair(h, insert_result.second);
 #else
         auto iter = subscription_set.find(key);
         if(iter == subscription_set.end()) {
             subscription_set.emplace(std::forward<K>(key), std::forward<V>(value));
             this->increase_subscriptions(h);
-            ++this->map_size;
+            this->increase_map_size();
         } else {
             iter->second = std::forward<V>(value);
         }
-        return std::make_pair(std::move(h), iter == subscription_set.end());
+        return std::make_pair(h, iter == subscription_set.end());
```

Comment:
Shouldn't this be `iter != subscription_set.end()` ?

Why did you stop moving the handle? You always return it, so accepting by const& is less optimial than accepting via value and returning via move.

Reply:
Return true if item was inserted. Move was removed accidental